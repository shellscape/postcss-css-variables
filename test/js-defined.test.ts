import { fixture, process } from './helpers';

const variables = {
  '--js-defined-important': {
    isImportant: true,
    value: '#0f0'
  },
  '--js-defined1': '75px',
  '--js-defined2': {
    value: '80px'
  },
  // Should be automatically prefixed with `--`
  'js-defined-no-prefix': '#ff0000'
};

describe('js-defined', async () => {
  test('work with JS defined variables', async () => {
    const css = await fixture('js-defined/js-defined');
    const result = await process(css, { variables });
    expect(result).toMatchSnapshot();
  });

  test('work with JS defined important variables', async () => {
    const css = await fixture('js-defined/js-defined-important');
    const result = await process(css, { variables });
    expect(result).toMatchSnapshot();
  });

  test('preserve -- declarations and var() values with `options.variables` AND `options.preserve`', async () => {
    const css = await fixture('js-defined/js-defined-preserve');
    const result = await process(css, { preserve: true, variables });
    expect(result).toMatchSnapshot();
  });

  test('preserve var() values and clean injected declarations with `options.variables` AND `options.preserve` AND `options.preserveInjectedVariables: false`', async () => {
    const css = await fixture('js-defined/js-defined-preserve-injected');
    const result = await process(css, {
      preserve: true,
      preserveInjectedVariables: false,
      variables
    });
    expect(result).toMatchSnapshot();
  });

  test('cast non-string values to string', async () => {
    const css = await fixture('js-defined/js-defined-non-string-values-casted-to-string');
    const result = await process(css, { variables });
    expect(result).toMatchSnapshot();
  });
});
