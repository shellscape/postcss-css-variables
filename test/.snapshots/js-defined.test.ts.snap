// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`js-defined > cast non-string values to string 1`] = `".box{top:undefined;width:undefined;height:undefined;overflow:undefined;font-size:undefined;visibility:undefined;opacity:undefined}"`;

exports[`js-defined > preserve -- declarations and var() values with \`options.variables\` AND \`options.preserve\` 1`] = `
":root{--js-defined-no-prefix:#ff0000}
:root{--js-defined2:80px}
:root{--js-defined1:75px}
:root{--js-defined-important:#0f0!important}
.box1{width:75px;width:var(--js-defined1);height:80px;height:var(--js-defined2);background:#ff0000;background:var(--js-defined-no-prefix)}"
`;

exports[`js-defined > preserve var() values and clean injected declarations with \`options.variables\` AND \`options.preserve\` AND \`options.preserveInjectedVariables: false\` 1`] = `
":root{}
:root{}
:root{}
:root{}
.box1{width:75px;width:var(--js-defined1);height:80px;height:var(--js-defined2);background:#ff0000;background:var(--js-defined-no-prefix)}"
`;

exports[`js-defined > work with JS defined important variables 1`] = `".box1{background-color:#0f0}"`;

exports[`js-defined > work with JS defined variables 1`] = `".box1{width:75px;height:80px;background:#ff0000}"`;
